src/cpp/astar.cpp

```diff
@@ -104,13 +145,15 @@ static PyObject *astar(PyObject *self, PyObject *args) {
         float new_cost = costs[cur.idx] + weights[nbrs[i]];
         if (new_cost < costs[nbrs[i]]) {
           // estimate the cost to the goal based on legal moves
-          if (diag_ok) {
-            heuristic_cost = linf_norm(nbrs[i] / w, nbrs[i] % w,
-                                       goal    / w, goal    % w);
-          }
-          else {
-            heuristic_cost = l1_norm(nbrs[i] / w, nbrs[i] % w,
-                                     goal    / w, goal    % w);
+          heuristic_cost = heuristic_func(nbrs[i] / w, nbrs[i] % w,
+                                          goal    / w, goal    % w);
+                                       
+          // add tiebreaker cost
+          if (tiebreaker_coefficient > 0) {
+            heuristic_cost = heuristic_cost + 
+                tiebreaker_coefficient/1000.0f * tie_breaker_func(nbrs[i] / w, nbrs[i] % w,
```

Comment:
Why pass the `tiebreaker_coefficient` as an `int` and not just pass this scaled-down `float` directly?

Reply:
Yeah, your right. Will change to float. 

Didn't want to pass a new type to c at the time.

Reply:
Update it to float. So now it is just 

if (tiebreaker_coefficient > 0.0f) {
    heuristic_cost = heuristic_cost + 
                tiebreaker_coefficient * tie_breaker_func(nbrs[i] / w, nbrs[i] % w,